### ISSUE TITLE
[tpch sqlserver]: date format

### ISSUE THREAD

**Issue by pmpetit** [TPC-Council/HammerDB#287]
Hello, trying to run tpch bench on sqlserver failed during schema creation.

The date "1993-MAY-10" is not recognize by sqlserver.

I tried to modify mssqlolap.tcl file,
to change the date format to YYYY-MM-DD, 
without success. 

If someone knows how to, i can try.

thx

**Comment by pmpetit** [comment]
for the moment, i changed all "date" to char(11) format instead of date

**Comment by sm-shaw** [comment]
Hi, 
What version of SQL Server are you using? What are the language settings? 
As this has not been raised before I would expect it to be a localisation issue. 
Note that also the datagen option enables more manipulation with loading by creating flat files first than can then be loaded with BULK INSERT https://www.hammerdb.com/docs/ch13s04.html

**Comment by pmpetit** [comment]
sqlserver 2019, language settings = FR.
Yes, i could use bulk insert to fill the db, but what about the qry run by the worklload ?, if they insert or update date type field, i will face the same error.

**Comment by sm-shaw** [comment]
Hi,
So yes, this looks like a language issue. There is no explicit support for languages other than English, however as an open source project of course a native speaker of any language is welcome to submit pull requests and provide support for a particular language if they wish to do so. 
Most of the queries are read-only, only one creates a view. The refresh function does insert data. 
another option could be adding the commands SET LANGUAGE or SET DATEFORMAT after login. 
This would be by adding a command such as:
odbc evaldirect "SET LANGUAGE English" to the build and/or driver script. 
if this is tested and works, a pull request could be submitted to add this to ensure the build works against installations with other languages.

**Comment by sm-shaw** [comment]
I have looked into this and am unable to replicate the error. I also cannot find a reference to the error message 
`"The date ...  is not recognized by sqlserver."` or an equivalent. 
I have tested with SQL Server partial containment with a pre-created database set to the French language and this does not error with the date format and the build is successful for both TPROC-H and TPROC-C.
It may be possible that SET LANGUAGE would resolve the issue, however reproducing is the first step and further guidance would be needed on reproducing then the issue if any changes are to be made.
